### PR TITLE
allow mdns cifs shares

### LINKED ISSUE #927: [feat]: allow raw ip addresses in the `hostname` field during cifs backups
##
Screenshot from 2021-12-09 16-37-23
Screenshot from 2021-12-09 17-02-51

# Prerequisites

- [X] I have searched for [existing issues](https://github.com/start9labs/embassy-os/issues) that already report this problem, without success.

### EmbassyOS Version

0.3.0 commit hash c2bb1e69be9878d93c471fafab780e603c244cfc

### Device

Laptop/Desktop

### Device OS

Linux

### Device OS Version

elementary OS 6

### Browser

Firefox

### Browser Version

94

### Current Behavior

When adding New Shared Folder hostname.local or IP address won't work.

### Expected Behavior

support hostnames and IP addresses 

### Steps to Reproduce

Just add your New Shared Folder

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dr-bonez**
closes #927 

cc @MattDHill

### CODE REVIEW

File: appmgr/src/disk/mount/filesystem/cifs.rs [hunk 1]
```diff
@@ -24,16 +25,36 @@ pub async fn mount_cifs(
...
-    .next()
-    .unwrap()
-    .parse()?;
+    let ip: IpAddr = if hostname.ends_with(".local") {
+        String::from_utf8(
+            Command::new("avahi-resolve-host-name")
+                .arg(hostname)
+                .invoke(crate::ErrorKind::Network)
+                .await?,
+        )?
```
Comment (ProofOfKeags) [inline appmgr/src/disk/mount/filesystem/cifs.rs:34]: why not use libavahi?

**Review by ProofOfKeags — APPROVED** [review]

File: appmgr/src/disk/mount/filesystem/cifs.rs (hunk 1 above)
Comment (ProofOfKeags) [reply appmgr/src/disk/mount/filesystem/cifs.rs:34]: factor out

**Comment by dr-bonez** [comment]
@k0gen please comment here with the results of your test

**Comment by k0gen** [comment]
> @k0gen please comment here with the results of your test

Fresh new sd-card flashed with eos.img 6942eda4323224055af3fae207c00ccc2c450ce3 build.

I toked the Migration route and wanted to recover from backup hosted on NAS. Entering `hostname.local` didn't work 

Embassy Setup 4

ended up entering just `hostname` and because the drive was not detected I restarted the setup and successfully migrated from the backup.

> Note the notification "Connection Failed" on iPadOS dark mode is hard to read.

I have also noticed that `About > LAN Address` is displaying https://embassy.local/

Screenshot from 2021-12-11 14-47-16

**Comment by k0gen** [comment]
Also forgot to mention this:

Screenshot from 2021-12-11 15-39-04

Notice those two are the same shares, only one has `/embassy` and the other `embassy` path.

**Pull request merged**